### PR TITLE
Uprev to Jinja2-3.0.0

### PR DESCRIPTION
**Pull request by tomhamiltonstubber**
Upgrading to the new version of Jinja2

https://github.com/pallets/jinja/blob/main/CHANGES.rst#version-300

### CODE REVIEW

File: harrier/extensions.py [hunk 1]
```diff
@@ -103,9 +96,9 @@ def load(self):
                     self._extensions[ext_type].extend([(path_match, attr) for path_match in attr.path_matches])
                 elif ext_type:
                     self._extensions[ext_type].append(attr)
-                elif any(getattr(attr, n, False) is True for n in filter_attrs):
+                elif getattr(attr, filter_attr, None):
```
Comment (samuelcolvin) [inline harrier/extensions.py:99]: is there a test for this case?

File: harrier/extensions.py [hunk 2]
```diff
@@ -103,9 +96,9 @@ def load(self):
...
                     self._extensions[ExtType.template_filters][attr_name] = attr
-                elif any(getattr(attr, n, False) is True for n in function_attrs):
+                elif getattr(attr, function_attr, None) or hasattr(attr, 'jinja_pass_arg'):
```
Comment (samuelcolvin) [inline harrier/extensions.py:101]: surely `jinja_pass_arg` should be added to `function_attrs` and this doesn't need changing?

Are there tests for context functions etc.?

Comment (tomhamiltonstubber) [reply harrier/extensions.py:101]: Why bother? Jinja 3.1 won't support the old methods so seems irrelevant.

File: harrier/extensions.py (hunk 1 above)
Comment (tomhamiltonstubber) [reply harrier/extensions.py:99]: Yep

Comment (tomhamiltonstubber) [reply harrier/extensions.py:99]: yep

File: harrier/extensions.py (hunk 2 above)
Comment (samuelcolvin) [reply harrier/extensions.py:101]: well it definitely doesn't make sense to have `function_attr` as a variable defined elsewhere and `'jinja_pass_arg'` as a string here, I don't mind which way around, but it should be consistent.

Comment (tomhamiltonstubber) [reply harrier/extensions.py]: Sure

**Pull request merged**